Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIrBuilder.kt

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+package org.jetbrains.kotlin.native.interop.gen
+
+import org.jetbrains.kotlin.native.interop.gen.jvm.InteropConfiguration
+import org.jetbrains.kotlin.native.interop.gen.jvm.KotlinPlatform
+import org.jetbrains.kotlin.native.interop.indexer.*
+
+/**
+ * Additional components that are required to generate bridges.
+ */
+interface BridgeGenerationComponents {
+
+    class GlobalSetterBridgeInfo(
+            val cGlobalName: String,
+            val typeInfo: TypeInfo
+    )
+
+    class GlobalGetterBridgeInfo(
+            val cGlobalName: String,
+            val typeInfo: TypeInfo,
+            val isArray: Boolean
+    )
+
+    val setterToBridgeInfo: Map<PropertyAccessor.Setter, GlobalSetterBridgeInfo>
+
+    val getterToBridgeInfo: Map<PropertyAccessor.Getter, GlobalGetterBridgeInfo>
+
+    val enumToTypeMirror: Map<ClassStub.Enum, TypeMirror>
+
+    val wCStringParameters: Set<FunctionParameterStub>
+
+    val cStringParameters: Set<FunctionParameterStub>
+}
+
+class BridgeGenerationComponentsBuilder(
+        val getterToBridgeInfo: MutableMap<PropertyAccessor.Getter, BridgeGenerationComponents.GlobalGetterBridgeInfo> = mutableMapOf(),
+        val setterToBridgeInfo: MutableMap<PropertyAccessor.Setter, BridgeGenerationComponents.GlobalSetterBridgeInfo> = mutableMapOf(),
+        val enumToTypeMirror: MutableMap<ClassStub.Enum, TypeMirror> = mutableMapOf(),
+        val wCStringParameters: MutableSet<FunctionParameterStub> = mutableSetOf(),
+        val cStringParameters: MutableSet<FunctionParameterStub> = mutableSetOf()
+) {
+    fun build(): BridgeGenerationComponents = object : BridgeGenerationComponents {
+        override val getterToBridgeInfo = this@BridgeGenerationComponentsBuilder.getterToBridgeInfo
```

Comment:
The common convention for builders and `build` methods is to copy mutable data.
E.g. by adding `.toMap()` here etc.